libs/community/langchain_community/utilities/pebblo.py

```diff
@@ -711,7 +722,9 @@ def prepare_docs_for_classification(
                     ),
                 }
             )
-        return docs, source_aggregate_size
+            if loader_details["loader"] == "SharePointLoader":
+                loader_details["source_path"] = doc_metadata.get("source_full_url")
+        return docs, source_aggregate_size, loader_details
```

Comment:
You don't need to return loader_details since it's updated directly in the original object.